Duplicate symbols (_TLSSslOptions and _last_parser) break build of syslog-ng 3.7.2 on OS X

Setup: OS X 10.11, Apple LLVM version 7.0.0 (clang-700.1.76)

After bypassing the issue mentioned in issue #806 by just making check_nanosleep() in timeutils.c return FALSE, I ran into a 'duplicate symbols' error when linking libsyslog-ng:
`````
Douglass-Mac-Pro:syslog-ng-3.7.2 dcarmich$ make
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory all-recursive
  CCLD     lib/libsyslog-ng.la
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
duplicate symbol _TLSSslOptions in:
    lib/.libs/lib_libsyslog_ng_la-tlscontext.o
    lib/transport/.libs/lib_libsyslog_ng_la-transport-tls.o
duplicate symbol _last_parser in:
    lib/.libs/lib_libsyslog_ng_la-cfg-grammar.o
    lib/parser/.libs/lib_libsyslog_ng_la-parser-expr-grammar.o
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libsyslog-ng.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```````

I applied that fix, but then it triggers two more issues:

`````
  GEN      lib/block-ref-grammar.y
  YACC     lib/block-ref-grammar.c
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison: invalid option -- W
Try `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison --help' for more information.
  YACC     lib/block-ref-grammar.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison: invalid option -- W
Try `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bison --help' for more information.
```````

```````
  CC       lib/lib_libsyslog_ng_la-block-ref-parser.lo
lib/block-ref-parser.c:26:10: fatal error: 'block-ref-grammar.h' file not found
#include "block-ref-grammar.h"
         ^
1 error generated.
````````

--- a/lib/parser/parser-expr-grammar.ym
+++ b/lib/parser/parser-expr-grammar.ym
@@ -36,8 +36,6 @@
 #include "plugin.h"
 #include "cfg-grammar.h"
 
-LogParser *last_parser;
-
 }
 
 %name-prefix "parser_expr_"

--- a/lib/tlscontext.h
+++ b/lib/tlscontext.h
@@ -44,7 +44,7 @@ typedef enum
   TVM_REQUIRED=0x0020,
 } TLSVerifyMode;
 
-enum
+typedef enum
 {
   TSO_NONE,
   TSO_NOSSLv2=0x0001,

--- a/modules/date/date-grammar.ym
+++ b/modules/date/date-grammar.ym
@@ -35,7 +35,6 @@
 #include "parser/parser-expr.h"
 #include "date-parser.h"
 
-LogParser *last_parser;
 }
 
 %name-prefix "date_"

--- a/modules/json/json-parser-grammar.ym
+++ b/modules/json/json-parser-grammar.ym
@@ -33,8 +33,7 @@
 #include "json-parser-grammar.h"
 #include "syslog-names.h"
 #include "messages.h"
-
-extern LogParser *last_parser;
+#include "cfg-grammar.h"
 
 }
 

--- a/modules/kvformat/kv-parser-grammar.ym
+++ b/modules/kvformat/kv-parser-grammar.ym
@@ -31,8 +31,7 @@
 #include "linux-audit-scanner.h"
 #include "cfg-parser.h"
 #include "kv-parser-grammar.h"
-
-extern LogParser *last_parser;
+#include "cfg-grammar.h"
 
 }
 

